### PR TITLE
Develop to main. Release v2

### PR DESCRIPTION
**Pull request by PavloNetrebchuk**
### Features
- Moved certificate access on course home ([#321](https://github.com/openedx/openedx-app-android/pull/321))
- Calendar main screen and dialogs ([#322](https://github.com/openedx/openedx-app-android/pull/322))
- Improved Dashboard level navigation ([#308](https://github.com/openedx/openedx-app-android/pull/308))
- Add external router for deep links ([#320](https://github.com/openedx/openedx-app-android/pull/320))
- Branch deep links to local calendar events ([#249](https://github.com/openedx/openedx-app-android/pull/249))
- Atlas push-pull scripts ([#317](https://github.com/openedx/openedx-app-android/pull/317))
- FCM integration for push notifications ([#344](https://github.com/openedx/openedx-app-android/pull/344))
- Handle Branch deep links from Braze push notifications ([#353](https://github.com/openedx/openedx-app-android/pull/353))
- FullStory Analytics SDK implementation ([#347](https://github.com/openedx/openedx-app-android/pull/347))
- Calendar synchronization ([#330](https://github.com/openedx/openedx-app-android/pull/330))
- xBlock offline mode ([#346](https://github.com/openedx/openedx-app-android/pull/346))
- Relative dates display ([#367](https://github.com/openedx/openedx-app-android/pull/367))
- Add flag to remove registration ([#387](https://github.com/openedx/openedx-app-android/pull/387))
- Update primary course on learn tab ([#389](https://github.com/openedx/openedx-app-android/pull/389))
- App-level error handling for WebView and network ([#392](https://github.com/openedx/openedx-app-android/pull/392))
- Course-level error handling for empty states ([#393](https://github.com/openedx/openedx-app-android/pull/393))
- Foundation library and analytics plugin ([#395](https://github.com/openedx/openedx-app-android/pull/395))

### Fixes
- Ensure cookies authentication before webview loads ([#312](https://github.com/openedx/openedx-app-android/pull/312))
- Crash on restoring app after inactivity ([#290](https://github.com/openedx/openedx-app-android/pull/290))
- Accessibility issue on course dashboard ([#327](https://github.com/openedx/openedx-app-android/pull/327))
- UI issues on auth screens ([#332](https://github.com/openedx/openedx-app-android/pull/332))
- Fix secondary course view button behavior ([#345](https://github.com/openedx/openedx-app-android/pull/345))
- Update keyboard visibility and IME action ([#350](https://github.com/openedx/openedx-app-android/pull/350))
- Video subtitles in native and YouTube player ([#349](https://github.com/openedx/openedx-app-android/pull/349))
- Missing email in Facebook login ([#377](https://github.com/openedx/openedx-app-android/pull/377))
- Crash when opening course from DashboardListFragment ([#336](https://github.com/openedx/openedx-app-android/pull/336))
- IllegalStateException in fragment transactions ([#373](https://github.com/openedx/openedx-app-android/pull/373))
- Various bug fixes from bug bash ([#390](https://github.com/openedx/openedx-app-android/pull/390))

### Chores & Maintenance
- Enhance app theme for edX branding ([#262](https://github.com/openedx/openedx-app-android/pull/262))
- Plural validation in strings.xml ([#348](https://github.com/openedx/openedx-app-android/pull/348))
- Remove Ukrainian translations from code ([#342](https://github.com/openedx/openedx-app-android/pull/342))
- Code refactoring and JUnit tests ([#391](https://github.com/openedx/openedx-app-android/pull/391))

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @PavloNetrebchuk!

### What's next?

*Please work through the following steps to get your changes ready for engineering review:*

#### :radio_button: Get product approval

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.

#### :radio_button: Provide context

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads





#### :radio_button: Get a green build

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.



#### :radio_button: Let us know that your PR is ready for review:

### Who will review my changes?


This repository is currently maintained by `@openedx/openedx-mobile-maintainers`.  Tag them in a comment and let them know that your changes are ready for review.


### Where can I find more information?

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)

### When can I expect my changes to be merged?

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*

**Review by volodymyr-chekyrta — APPROVED** [review]

**Pull request merged**
